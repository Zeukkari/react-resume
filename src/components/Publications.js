import React from 'react'

import { Box, Paragraph, Heading, Text } from 'grommet'

import { ResumeSection, SectionTitle } from './common'

const Publications = ({ publications }) => {
  const renderItems = publications.map(item => {
    return (
      <Box key="item.company">
        <Heading level={3}>{item.name}</Heading>
        <Text weight="bold" size="medium">
          {item.publisher}
        </Text>
        <Text>{item.releaseDate}</Text>
        <Paragraph>{item.summary}</Paragraph>
      </Box>
    )
  })
  return (
    <Box margin="none" pad="none" border="all">
      <ResumeSection>
        <SectionTitle>Publications</SectionTitle>
        <Box>{renderItems}</Box>
      </ResumeSection>
    </Box>
  )
}

Publications.defaultProps = {
  publications: [],
}

export default Publications
