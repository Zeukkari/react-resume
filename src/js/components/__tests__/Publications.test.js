import React from 'react'
import renderer from 'react-test-renderer'

import {Publications} from '../Publications'

import defaultResume from './mockResume.json'

test('renders publications section', () => {
  const data = defaultResume.publications
  const component = renderer.create(<Publications publications={data} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
