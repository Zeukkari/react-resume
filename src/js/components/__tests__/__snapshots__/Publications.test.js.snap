// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders publications section 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 eRCNqM"
>
  <div
    className="StyledBox-sc-13pk1d4-0 lhNMsi"
  >
    <h1
      className="StyledHeading-sc-1rdh4aw-0 jVpYfH"
      size="medium"
    >
      Publications
    </h1>
    <div
      className="StyledBox-sc-13pk1d4-0 IRSNj"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 IRSNj"
      >
        <h3
          className="StyledHeading-sc-1rdh4aw-0 gRvHvF"
        >
          Publication
        </h3>
        <span
          className="StyledText-sc-1sadyjn-0 dndifk"
          size="medium"
        >
          Company
        </span>
        <span
          className="StyledText-sc-1sadyjn-0 fWSbXS"
        >
          2014-10-01
        </span>
        <p
          className="StyledParagraph-tbetod-0 kxFtnQ"
        >
          Description...
        </p>
      </div>
    </div>
  </div>
</div>
`;
